platform/darwin/src/MGLLocationManager.m

```diff
@@ -0,0 +1,108 @@
+#import "MGLLocationManager_Private.h"
+
+@interface MGLCLLocationManager()<CLLocationManagerDelegate>
+
+@property (nonatomic) CLLocationManager *locationManager;
+
+@end
+
+@implementation MGLCLLocationManager
+
+- (instancetype)init
+{
+    if (self = [super init]) {
+        _locationManager = [[CLLocationManager alloc] init];
+        _locationManager.delegate = self;
```

Comment:
We'll need a `dealloc` with `_locationManager.delegate = nil;`

`CLLocationManager`'s delegate property is declared as `assign`, rather than `weak`.

Reply:
Oh! nvm you are saying that the `dealloc` is missing a `_locationManager.delegate = nil;`

Reply:
👍